src/core/webview/ClineProvider.ts

```diff
@@ -890,14 +884,17 @@ export class ClineProvider extends EventEmitter<ClineProviderEvents> implements
 		return getSettingsDirectoryPath(globalStoragePath)
 	}
 
-	// not private, so it can be accessed from webviewMessageHandler
-	async ensureCacheDirectoryExists() {
+	private async ensureCacheDirectoryExists() {
 		const { getCacheDirectoryPath } = await import("../../shared/storagePathManager")
 		const globalStoragePath = this.contextProxy.globalStorageUri.fsPath
 		return getCacheDirectoryPath(globalStoragePath)
 	}
 
-	// not private, so it can be accessed from webviewMessageHandler
+	async writeModelsToCache<T>(filename: string, data: T) {
```

Comment:
New function, avoids need to make `ensureCacheDirectoryExists` public, and offers a higher level of abstraction to message handlers.

Symmetry with pre-existing `readModelsFromCache` function.

Reply:
Maybe in the future we can remove file management from this file.